Bukkit/src/main/java/com/plotsquared/bukkit/queue/LimitedRegionWrapperQueue.java

```
@@ -0,0 +1,114 @@
+import com.sk89q.jnbt.CompoundTag;
+import com.sk89q.worldedit.bukkit.BukkitAdapter;
+import com.sk89q.worldedit.entity.Entity;
+import com.sk89q.worldedit.function.pattern.Pattern;
+import com.sk89q.worldedit.math.BlockVector3;
+import com.sk89q.worldedit.world.block.BaseBlock;
+import com.sk89q.worldedit.world.block.BlockState;
+import org.bukkit.Location;
+import org.bukkit.entity.EntityType;
+import org.bukkit.generator.LimitedRegion;
+import org.checkerframework.checker.nullness.qual.NonNull;
+
+/**
+ * Wraps a {@link LimitedRegion} inside a P2 {@link com.plotsquared.core.queue.QueueCoordinator} so it can be written to.
+ *
+ * @since TODO
+ */
+public class LimitedRegionWrapperQueue extends DelegateQueueCoordinator {
+
+    private final LimitedRegion limitedRegion;
+
+    /**
+     * @since TODO
+     */
+    public LimitedRegionWrapperQueue(LimitedRegion limitedRegion) {
+        super(null);
+        this.limitedRegion = limitedRegion;
+    }
+
+    @Override
+    public boolean setBlock(final int x, final int y, final int z, @NonNull final Pattern pattern) {
+        return setBlock(x, y, z, pattern.applyBlock(BlockVector3.at(x, y, z)));
+    }
+
+    @Override
+    public boolean setBlock(final int x, final int y, final int z, @NonNull final BaseBlock id) {
+        boolean result = setBlock(x, y, z, id.toImmutableState());
+        if (result && id.hasNbtData()) {
+            CompoundTag tag = id.getNbtData();
+            StateWrapper sw = new StateWrapper(tag);
+            sw.restoreTag(limitedRegion.getBlockState(x, y, z).getBlock());
+        }
+        return result;
+    }
+
+    @Override
+    public boolean setBlock(final int x, final int y, final int z, @NonNull final BlockState id) {
+        try {
+            limitedRegion.setType(x, y, z, BukkitAdapter.adapt(id.getBlockType()));
+            limitedRegion.setBlockData(x, y, z, BukkitAdapter.adapt(id));
+        } catch (IllegalArgumentException e) {
+            e.printStackTrace();
```

If we can ignore the exception, this should be removed. If we can't ignore it, it should be logged via a logger instead.